pkg/services/publicdashboards/api/api.go

```diff
@@ -117,23 +117,28 @@ func (api *Api) GetPublicDashboardConfig(c *models.ReqContext) response.Response
 
 // sets public dashboard configuration for dashboard
 func (api *Api) SavePublicDashboardConfig(c *models.ReqContext) response.Response {
+	// exit if we don't have a valid dashboardUid
+	dashboardUid := web.Params(c.Req)[":uid"]
+	if dashboardUid == "" {
+		handleDashboardErr(http.StatusBadRequest, "no dashboardUid", dashboards.ErrDashboardIdentifierNotSet)
+	}
+
```

Comment:
If we are not making any specific validation on this `uid` (like the length for example) this validation will never return an error because it is a part of the url, if it is not there then this handler won't be executed.